In [1]:
import pandas as pd
import json
from pandas import json_normalize
import requests
import time

Get all comments between January 1, 2019 at 00:00:00 (1546300800)
and December 31, 2020 at 23:59:59 (1577836799)
`
Mini test: until January 7, 2019 at 23:59:59 (1546905599)

In [2]:
df_list = []
iteration_count = 0

In [3]:
current_time = 1546300800 # failed earlier at this time, retrying
end_time = 1546905599 # January 7
# subreddit = "wallstreetbets"
# size = 1000

link_holder = "https://api.pushshift.io/reddit/search/comment/?subreddit=wallstreetbets&after={}&before={}&size=1000"

In [5]:
while current_time < end_time:
    r = requests.get(link_holder.format(current_time, end_time))
        
    if r.json()['data'] == []:
        print('empty')
        break
    df = json_normalize(r.json(), "data")
    df_list.append(df)
    
    # get the newest comment in the data frame and set it to the current time 
    # so next time we get a new set of 500 comments
    
    time.sleep(.4)
    current_time = df["created_utc"].max()
    if iteration_count % 20 == 0:
        print(iteration_count, current_time)
    else:
        print(iteration_count, current_time, end=" ")
    iteration_count += 1

0 1546362502
1 1546383218 2 1546397988 3 1546409687 4 1546436345 5 1546443079 6 1546447997 7 1546455233 8 1546460306 9 1546465345 10 1546468573 11 1546474078 12 1546480870 13 1546489533 14 1546498397 15 1546518245 16 1546526731 17 1546530348 18 1546534288 19 1546538838 20 1546543548
21 1546547545 22 1546551207 23 1546557924 24 1546566400 25 1546575148 26 1546592499 27 1546611407 28 1546615472 29 1546618558 30 1546622636 31 1546626869 32 1546631124 33 1546635453 34 1546642839 35 1546655603 36 1546677628 37 1546711911 38 1546731369 39 1546754521 40 1546793413
41 1546810768 42 1546823666 43 1546835246 44 1546867339 45 1546875560 46 1546881004 47 1546886861 48 1546892894 49 1546901084 50 1546905595 empty


In [18]:
comments = pd.concat(df_list, ignore_index=True)
comments

author  author_created_utc author_flair_background_color  \
0                Nm0369        1.493507e+09                          None   
1           achennupati        1.487615e+09                          None   
2             grissomza        1.443582e+09                          None   
3            ktdfintech        1.528505e+09                          None   
4      Lester_Diamond23        1.503953e+09                          None   
...                 ...                 ...                           ...   
51491       bobvonbobby        1.433353e+09                          None   
51492            Kraka2        1.485284e+09                          None   
51493     BaydinMemeler        1.531276e+09                          None   
51494       Long_Flower        1.531416e+09                          None   
51495          Marshy92        1.422774e+09                          None   

      author_flair_css_class author_flair_richtext author_flair_template_id  \
0                       None                    []                     None   
1                       None                    []                     None   
2                       None                    []                     None   
3                       None                    []                     None   
4                       None                    []                     None   
...                      ...                   ...                      ...   
51491                   None                    []                     None   
51492                   None                    []                     None   
51493                   None                    []                     None   
51494                   None                    []                     None   
51495                   None                    []                     None   

      author_flair_text author_flair_text_color author_flair_type  \
0                  None                    None              text   
1                  None                    None              text   
2                  None                    None              text   
3                  None                    None              text   
4                  None                    None              text   
...                 ...                     ...               ...   
51491              None                    None              text   
51492              None                    None              text   
51493              None                    None              text   
51494              None                    None              text   
51495              None                    None              text   

      author_fullname  ... send_replies stickied       subreddit subreddit_id  \
0           t2_17iqgr  ...         True    False  wallstreetbets     t5_2th52   
1           t2_15l23j  ...         True    False  wallstreetbets     t5_2th52   
2            t2_qtsq4  ...         True    False  wallstreetbets     t5_2th52   
3         t2_1jas53nt  ...         True    False  wallstreetbets     t5_2th52   
4          t2_c14dcw5  ...         True    False  wallstreetbets     t5_2th52   
...               ...  ...          ...      ...             ...          ...   
51491        t2_nvy2m  ...         True    False  wallstreetbets     t5_2th52   
51492       t2_14qpwe  ...         True    False  wallstreetbets     t5_2th52   
51493     t2_1qxsd4bn  ...         True    False  wallstreetbets     t5_2th52   
51494      t2_ovx1yx0  ...         True    False  wallstreetbets     t5_2th52   
51495        t2_l2x2n  ...         True    False  wallstreetbets     t5_2th52   

      updated_utc  gildings.gid_1 gildings.gid_2 gildings.gid_3  \
0      1546300808               0              0              0   
1      1546300809               0              0              0   
2      1546300824               0              0              0   
3      1546300854               0              0              0   
4     

In [19]:
len(comments)

51496

In [20]:
current_time

1546905595

In [21]:
# the first 500 comments were spread over a time of 9050 seconds 
comments['created_utc'].max() 

604795

In [30]:
display(comments['created_utc'].sort_values()[1000] - 1546300800)

(end_time - 1546300800) / iteration_count

21662

11858.803921568628

The first request gave us 21662 seconds worth of comments (360 minute)

On average each request gave us 11858 seconds worth of comments comments (almost 200 minutes)